tests/hazmat/primitives/test_dh.py

```diff
@@ -74,31 +84,268 @@ def test_dh_numbers():
+                  "676809305984951446"),
+         "y": int("7991113599078333875847794507844575"
+                  "6175050224504463981970753203231868"
+                  "0136551315101498332139075360575330"
+                  "6200081479128423955529753566986512"
+                  "451925943417370573")},
+        {"p": int("8230910504630293046781610423436023"
+                  "7855700588139834267641276801633860"
+                  "7909748155624229540803040656254044"
+                  "1574901920011043362099679138268556"
+                  "410914150106172603"),
+         "g": 2,
+         "y": int("3153667608586790035405826466812972"
+                  "2719824210153531748906003251088123"
+                  "2786742480574293420585405275184080"
+                  "7361048259220540363086632109209018"
+                  "127565889223022306"),
+         "x": int("5094233318592725701435923671489794"
+                  "1349417222942424283646053422521046"
+                  "7803823560524571564089891180147290"
+                  "6659807096135548788171923289152060"
+                  "442445002898139883")})
+    ]
+
+    @pytest.mark.parametrize("params_pair", bad_tls_exchange_params)
+    def test_bad_tls_exchange(self, backend, params_pair):
+        parameters1 = dh.DHParameterNumbers(params_pair[0]["p"],
+                                            params_pair[0]["g"])
+        public1 = dh.DHPublicNumbers(params_pair[0]["y"], parameters1)
+        private1 = dh.DHPrivateNumbers(params_pair[0]["x"], public1)
+        key1 = private1.private_key(backend)
+        pub_key1 = key1.public_key()
+
+        parameters2 = dh.DHParameterNumbers(params_pair[1]["p"],
+                                            params_pair[1]["g"])
+        public2 = dh.DHPublicNumbers(params_pair[1]["y"], parameters2)
+        private2 = dh.DHPrivateNumbers(params_pair[1]["x"], public2)
+        key2 = private2.private_key(backend)
+        pub_key2 = key2.public_key()
+
+        if pub_key2.public_numbers().y >= parameters1.p:
+            with pytest.raises(ValueError):
+                key1.exchange(pub_key2)
+        else:
+            symkey1 = key1.exchange(pub_key2)
+            assert symkey1
+
+            symkey2 = key2.exchange(pub_key1)
+
+            assert symkey1 != symkey2
+
+    @pytest.mark.parametrize(
+        "vector",
+        load_vectors_from_file(
+            os.path.join("asymmetric", "DH", "vec.txt"),
+            load_nist_vectors))
+    def test_dh_vectors(self, backend, vector):
+        parameters = dh.DHParameterNumbers(int(vector["p"]),
+                                           int(vector["g"]))
+        public = dh.DHPublicNumbers(int(vector["y"]), parameters)
+        private = dh.DHPrivateNumbers(int(vector["x"]), public)
+        key = private.private_key(backend)
+        symkey = key.exchange(public.public_key(backend))
+
+        assert int_from_bytes(symkey, 'big') == int(vector["k"], 16)
```

Comment:
I can do this but I think this way is better because of the padding. In case that any of the vectors need padding your test will fail because the symkey buffer is padded.